File: app/src/main/java/hi_javafx/FXMLController.java
package hi_javafx;

import java.net.URL;
import java.util.ResourceBundle;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.control.Label;
import javafx.scene.control.Button;

public class FXMLController implements Initializable {
    
    @FXML
    private Label label2;

    @FXML
    private Button button;

    @FXML
    private void handleButtonAction(ActionEvent event) {
        System.out.println("You clicked me! + " + button.getText());
        label2.setText("Hello World!");
    }
    
    @Override
    public void initialize(URL url, ResourceBundle rb) {
        System.out.println(
            "relative to root: " + url + 
            " localize: " + rb);
    }    
}

File: app/src/main/kotlin/FXMLController.kt
package hi_javafx

import java.net.URL
import java.util.ResourceBundle
import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.control.Label
import javafx.scene.control.Button

class FXMLControllerKt () : Initializable
{

    @FXML 
    lateinit private var label: Label

    @FXML
    lateinit private var button: Button

    @FXML
    @Suppress("UNUSED_PARAMETER")
    private fun handleButtonAction (event: ActionEvent)
    {
        System.out.println("Button was clicked!")
        label.setText("Hello JavaFX with Kotlin")
    }

    override 
    fun initialize(url: URL?, bundle: ResourceBundle?) 
    { 
        System.out.println("url: $url, bundle: $bundle")
    }
}